### DISCUSSION TITLE
Sequelize query data between columns

### DISCUSSION THREAD

**Discussion by EmmanJayDizon** [sequelize/sequelize#15290] [Help & Questions]
Is this possible on sequelize query without using raw query ? ->

SELECT *
FROM tablename
WHERE 10000 BETWEEN "minAmount" AND "maxAmount"
AND 30 BETWEEN "minDay" AND "maxDay"

**Comment by ephys** [comment]
This should work:

```ts
import { where, Op, col } from 'sequelize';

YourModel.findAll({
  where: and(
    where(10000, Op.between, [col("minAmount"), col("maxAmount")]),
    where(30, Op.between, [col("minDay"), col("maxDay")])
  )
});
```

**Reply by EmmanJayDizon** [reply]
Hi @ephys, didn't work for me. Btw im using commonJS. here's what i've done



```
const { and, where, Op, col } = require('sequelize');

MyModel.findOne({
      where: and(
        where(10000, Op.between, [col('minAmount'), col('maxAmount')]),
        where(30, Op.between, [col('minDay'), col('maxDay')]),
      ),
    });
```
    I've got an error -> " TypeError: Cannot read properties of undefined (reading 'name')"


node version -> v16.17.0
sequelize version -> "^6.25.6",

**Reply by EmmanJayDizon** [reply]
I did this also but same error " TypeError: Cannot read properties of undefined (reading 'name')"

```
const { Op, where, col } = require('sequelize');

this.model.findOne({
      where: {
        [Op.and]: [
          where(10000, Op.between, [col('minAmount'), col('maxAmount')]),
          where(30, Op.between, [col('minDay'), col('maxDay')]),
        ],
      },
    });
```

**Reply by ephys** [reply]
That's unfortunate. Something we'll need to fix. I'll add a test for it

In the meantime, this syntax should work:

```ts
YourModel.findAll({
  where: and(
    literal('10000 BETWEEN "minAmount" AND "maxAmount"'),
    literal('30 BETWEEN "minDay" AND "maxDay"'),
  ),
});
```

or if you need variables:

```ts
YourModel.findAll({
  where: and(
    literal(':amount BETWEEN "minAmount" AND "maxAmount"'),
    literal(':day BETWEEN "minDay" AND "maxDay"'),
  ),
  replacements: { amount: 10000, day: 30 },
});
```

Making literals better to use is also something we're working on https://github.com/sequelize/sequelize/issues/14299

**Reply by EmmanJayDizon** [reply]
Thankyou. This works for me now.
I'm just looking for another decent solution on this one :


   ```
    minAmount: {
          [Op.lte]: amount,
        },
        maxAmount: {

          [Op.gte]: amount,
        },
        minDay: {
          [Op.lte]: tenor,
        },
        maxDay: {
          [Op.gte]: tenor,
        },
```

that's why.

Kindly inform me once you already fix the issue on our previous discussion.

Thankyou very much !

**Reply by ephys** [reply]
You can subscribe to this PR if you want :) https://github.com/sequelize/sequelize/pull/14020. It's the "fix all the where bugs" PR
